### PR TITLE
[2388] Consolidated icons overlay management into a single component

### PR DESCRIPTION
**Pull request by frouene**
Bug: https://github.com/eclipse-sirius/sirius-web/issues/2388

# Pull request template

## General purpose
What is the main goal of this pull request?
- [ ] Bug fixes
- [ ] New features
- [ ] Documentation
- [x] Cleanup
- [ ] Tests
- [ ] Build / releng

## Project management
- [ ] Has the pull request been added to the relevant project and milestone? (Only if you know that your work is part of a specific iteration such as the current one)
- [ ] Have the `priority:` and `pr:` labels been added to the pull request? (In case of doubt, start with the labels `priority: low` and `pr: to review later`)
- [ ] Have the relevant issues been added to the pull request?
- [ ] Have the relevant labels been added to the issues? (`area:`, `difficulty:`, `type:`)
- [ ] Have the relevant issues been added to the same project and milestone as the pull request?
- [ ] Has the `CHANGELOG.adoc` been updated to reference the relevant issues?
- [ ] Have the relevant API breaks been described in the `CHANGELOG.adoc`? (Including changes in the GraphQL API)
- [ ] In case of a change with a visual impact, are there any screenshots in the `CHANGELOG.adoc`? For example in `doc/screenshots/2022.5.0-my-new-feature.png`

## Architectural decision records (ADR)
- [ ] Does the title of the commit contributing the ADR start with `[doc]`?
- [ ] Are the ADRs mentioned in the relevant section of the  `CHANGELOG.adoc`?

## Dependencies
- [ ] Are the new / upgraded dependencies mentioned in the relevant section of the `CHANGELOG.adoc`?
- [ ] Are the new dependencies justified in the `CHANGELOG.adoc`?

## Frontend

This section is not relevant if your contribution does not come with changes to the frontend.

### General purpose
- [ ] Is the code properly tested? (Plain old JavaScript tests for business code and tests based on React Testing Library for the components)

### Typing
We need to improve the typing of our code, as such, we require every contribution to come with proper TypeScript typing for both changes contributing new files and those modifying existing files.
Please ensure that the following statements are true for each file created or modified (this may require you to improve code outside of your contribution).

- [ ] Variables have a proper type
- [ ] Functions’ arguments have a proper type
- [ ] Functions’ return type are specified
- Hooks are properly typed:
	- [ ] `useMutation<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useQuery<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useSubscription<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useMachine<CONTEXT_TYPE, EVENTS_TYPE>(…)`
	- [ ] `useState<STATE_TYPE>(…)`
- [ ] All components have a proper typing for their props
- [ ] No useless optional chaining with `?.` (if the GraphQL API specifies that a field cannot be `null`, do not treat it has potentially `null` for example)
- [ ] Nullable values have a proper type (for example `let diagram: Diagram | null = null;`)

## Backend

This section is not relevant if your contribution does not come with changes to the backend.

### General purpose
- [ ] Are all the event handlers tested?
- [ ] Are the event processor tested?
- [ ] Is the business code (services) tested?
- [ ] Are diagram layout changes tested?

### Architecture
- [ ] Are data structure classes properly separated from behavioral classes?
- [ ] Are all the relevant fields final?
- [ ] Is any data structure mutable? If so, please write a comment indicating why
- [ ] Are behavioral classes either stateless or side effect free?

### How to test this PR?
_Please describe here the various use cases to test this pull request_

- [ ] Has the Kiwi TCMS test suite been updated with tests for this contribution?

### CODE REVIEW

File: packages/core/frontend/sirius-components-core/src/icon/IconOverlay.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
+/*******************************************************************************
+ * Copyright (c) 2023, 2023 Obeo.
```
Comment (pcdavid) [inline packages/core/frontend/sirius-components-core/src/icon/IconOverlay.tsx]: 2023 twice

File: packages/forms/frontend/sirius-components-forms/src/groups/Group.tsx [hunk 2]
```diff
@@ -69,17 +69,6 @@ const useGroupStyles = makeStyles<Theme, GroupStyleProps>((theme) => ({
     paddingTop: '1px',
     paddingBottom: '0px',
   },
-  buttonIcon: {
-    marginRight: theme.spacing(1),
```
Comment (pcdavid) [inline packages/forms/frontend/sirius-components-forms/src/groups/Group.tsx:73]: We're losing the `marginRight: theme.spacing(1)`. Is that correct?

File: packages/forms/frontend/sirius-components-forms/src/propertysections/ListPropertySection.tsx [hunk 3]
```diff
@@ -101,17 +101,6 @@ const useListPropertySectionStyles = makeStyles<Theme, ListStyleProps>((theme) =
     whiteSpace: 'nowrap',
     flexGrow: 1,
   },
-  iconContainer: {
-    position: 'relative',
-    width: '16px',
-    height: '16px',
-    marginRight: theme.spacing(2),
```
Comment (pcdavid) [inline packages/forms/frontend/sirius-components-forms/src/propertysections/ListPropertySection.tsx:108]: We're losing the `marginRight`.

File: packages/forms/frontend/sirius-components-forms/src/propertysections/__tests__/__snapshots__/ListPropertySection.test.tsx.snap [hunk 4]
```diff
@@ -905,3 +905,153 @@ exports[`should the click event sent on item click 2`] = `
   </div>
 </div>
 `;
+
+exports[`should the click event sent on item click 3`] = `
```
Comment (pcdavid) [inline packages/forms/frontend/sirius-components-forms/src/propertysections/__tests__/__snapshots__/ListPropertySection.test.tsx.snap]: Is this addition normal?

File: packages/forms/frontend/sirius-components-widget-reference/src/components/ValuedReferenceAutocomplete.tsx [hunk 5]
```diff
@@ -271,20 +247,9 @@ export const ValuedReferenceAutocomplete = ({
...
-                      width="16"
-                      key={index}
-                      alt={option.kind}
-                      src={httpOrigin + icon}
-                      className={classes.icon}
-                      style={{ zIndex: index }}></img>
-                  ))}
-                </div>
-              )
+              <div>
```
Comment (pcdavid) [inline packages/forms/frontend/sirius-components-widget-reference/src/components/ValuedReferenceAutocomplete.tsx:250]: Is the `div` necessary?

File: packages/selection/frontend/sirius-components-selection/src/SelectionDialog.tsx [hunk 6]
```diff
@@ -162,19 +151,7 @@ export const SelectionDialog = ({
                 data-testid={selectionObject.label}>
                 <ListItemIcon>
                   {selectionObject.iconURL.length > 0 ? (
-                    <div className={classes.iconContainer}>
-                      {selectionObject.iconURL.map((icon, index) => (
-                        <img
-                          height="24"
-                          width="24"
```
Comment (pcdavid) [inline packages/selection/frontend/sirius-components-selection/src/SelectionDialog.tsx:169]: This is changing the icon size from 24×24 to 16×16.

**Review by pcdavid — CHANGES_REQUESTED** [review]

File: packages/core/frontend/sirius-components-core/src/icon/IconOverlay.tsx (hunk 1 above)
Comment (frouene) [reply packages/core/frontend/sirius-components-core/src/icon/IconOverlay.tsx]: done

File: packages/forms/frontend/sirius-components-forms/src/groups/Group.tsx (hunk 2 above)
Comment (frouene) [reply packages/forms/frontend/sirius-components-forms/src/groups/Group.tsx:73]: I added the possibility to custom icon style.
Done

File: packages/forms/frontend/sirius-components-forms/src/propertysections/ListPropertySection.tsx (hunk 3 above)
Comment (frouene) [reply packages/forms/frontend/sirius-components-forms/src/propertysections/ListPropertySection.tsx:108]: Done

File: packages/selection/frontend/sirius-components-selection/src/SelectionDialog.tsx (hunk 6 above)
Comment (frouene) [reply packages/selection/frontend/sirius-components-selection/src/SelectionDialog.tsx:169]: I added the possibility to custom icon size.
Done

File: packages/forms/frontend/sirius-components-widget-reference/src/components/ValuedReferenceAutocomplete.tsx (hunk 5 above)
Comment (frouene) [reply packages/forms/frontend/sirius-components-widget-reference/src/components/ValuedReferenceAutocomplete.tsx:250]: Yes, MUI add some chip style related properties to the div only if the div is at this level.

**Review by pcdavid — APPROVED** [review]

**Comment by pcdavid** [comment]
Waiting for the build to merge.

**Pull request merged**